hazelcast/src/main/java/com/hazelcast/cp/internal/datastructures/spi/atomic/RaftAtomicValueService.java

```diff
@@ -204,4 +205,14 @@ protected final Map<CPGroupId, Object> getSnapshotMap(int partitionId) {
     protected final void clearPartitionReplica(int partitionId) {
         atomicValues.keySet().removeIf(t -> raftService.getCPGroupPartitionId(t.element1) == partitionId);
     }
+
+    Set<BiTuple<CPGroupId, String>> getDestroyedValues() {
+        return destroyedValues;
+    }
+
+    public void clearDestroyedValues(CPGroupId cpGroupId) {
+        List<BiTuple<CPGroupId, String>> toWipe =
+                destroyedValues.stream().filter(tuple -> tuple.element1.equals(cpGroupId)).collect(Collectors.toList());
+        toWipe.forEach(destroyedValues::remove);
```

Comment:
What about `destroyedValues.removeAll(toWipe)`?

Reply:
Doesn't matter -- actually, the `foreach` can be preferable in some scenarios where `removeAll` takes a long time to complete.